### PR TITLE
Add tests to es6 challenge Write Higher Order Arrow Functions.

### LINKED ISSUE #14833: The test is completed without writing the arrow function.
Challenge [write-higher-order-arrow-functions](http://beta.freecodecamp.com/en/challenges/es6/write-higher-order-arrow-functions) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.96 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

```js

"use strict";
const realNumberArray = [4, 5.6, -9.8, 3.14, 42, 6, 8.34];
// change code below this line

var squaredIntegers = realNumberArray;
// change code above this line
// test your code
console.log(squaredIntegers);

```

### PR DESCRIPTION
**Pull request by MiloATH**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Helps to close #14833

#### Description

This adds tests cases to es6 challenge Write Higher Order Arrow Functions. There were commented tests that needed to be implemented. I add only the tests which were specified. I don't believe there is a way to check if an element is a const or a var since the inputted code is changed. The `code` object translates const and let into var. This is likely because not all browsers allow const and let. This means a regex can't be used to check if the user is using a const and not a var. I left those tests commented out.

### CODE REVIEW

**Comment by raisedadead** [comment]
@MiloATH thanks a lot for adding these tests, just to let you know, we have some changes upcoming via https://github.com/freeCodeCamp/freeCodeCamp/pull/15041 and https://github.com/freeCodeCamp/freeCodeCamp/pull/14707

I am okay to merge your changes pending a go-ahead from @freeCodeCamp/moderators especially thoughts from @BerkeleyTrue and @no-stack-dub-sack

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (no-stack-dub-sack) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: @MiloATH This test will also not work at present, for the same reason as above. The arrow syntax gets transpiled into ES5 code just like `let` and `const`.

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (no-stack-dub-sack) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: @MiloATH Also, when adding tests, we should try our best to keep everything formatted correctly, so that we will have to go over these less times to get them in good shape. In this case, what's missing is `<code>` tags surrounding keywords  and code snippets in your test case messages, i.e. `map`, `filter`, `=>` (even though I think that test should probably remain commented out for now), `function`, `squaredIntegers`, `[16, 1764, 36]`, etc. 

This isn't the biggest deal in the world, but it helps keep things clean and consistent across the site, and ultimately makes for a better user experience. 

So final assessment, if you could:
- comment out both tests again involving the `function` keyword and `=>` syntax until we have working support for ES6 tests
- fix formatting issues

Once this is done, I think we'd be fine to merge this in since it does not involve any ES6 tests, not a bad idea to fill in the blanks where we can until ES6 support is added. Thanks for your help with this!

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (no-stack-dub-sack) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: @MiloATH This test will not work currently, since the `=>` syntax gets transpiled into the `function` keyword. Before making a PR, the safest thing to do is test each of these locally to make sure they work.

**Review by no-stack-dub-sack — CHANGES_REQUESTED** [review]
@MiloATH I think for the time being, the test regarding the arrow function should be left

**Comment by no-stack-dub-sack** [comment]
@raisedadead @MiloATH Some of these test cases still will not work due to our current limitations. I'm not sure if the changes in #15041 will actually be going through, but I am trying to figure out another way to add ES6 support soon hopefully. 

@BerkeleyTrue suggested that we may already have that support, and if so, we may need to make some minor tweaks and communicate them to anyone willing to write these challenges, but until then, we should hold off on introducing test cases that will not work - that will just open a pandoras box of new issues on the repo.

**Comment by no-stack-dub-sack** [comment]
@MiloATH changes look good, thanks! Can you just squash your commits before we merge this in?

**Review by no-stack-dub-sack — APPROVED** [review]

**Comment by no-stack-dub-sack** [comment]
## LGTM! 

Thanks @MiloATH!

**Pull request merged**
